test/servers/basic.js

```diff
@@ -14,4 +15,5 @@ const server = http.createServer(function (req, res) {
   }
 })
 
+try { fs.unlinkSync(sock) } catch (err) {}
```

Comment:
Might it be worth logging something to the console here in the unexpected case? So that if we get other errors in that hard-to-reproduce case, now or with future changes, there is a clue.

If so, might be worth moving this code to a util.js or something and sharing it between the test servers rather than duplicating it.